src/tests/ftest/util/ServerUtils.py

```diff
@@ -33,18 +34,110 @@
+    """
+    Enable/Disable NVMe Mode.
+    By default it's Enabled in yaml file so disable to run with ofi+sockets.
+    """
+    if 'bdev_class' in default_value['servers'][0]:
+        if (default_value['servers'][0]['bdev_class'] == bdev and
+                not enabled):
+            del default_value['servers'][0]['bdev_class']
+    if enabled:
+        default_value['servers'][0]['bdev_class'] = bdev
+
+def remove_mux_from_yaml(avocado_yaml_file, avocado_yaml_file_tmp):
+    """
+    Function to remove "!mux" from yaml file and create new tmp file
+    Args:
+        avocado_yaml_file: Avocado test yaml file
+        avocado_yaml_file_tmp: Avocado temporary test yaml file
+    """
+    with open(avocado_yaml_file, 'r') as rfile:
+        filedata = rfile.read()
+    filedata = filedata.replace('!mux', '')
+    with open(avocado_yaml_file_tmp, 'w') as wfile:
+        wfile.write(filedata)
+
+def create_server_yaml(basepath):
+    """
+    This function is to create DAOS server configuration YAML file
+    based on Avocado test Yaml file.
+    Args:
+        - basepath = DAOS install basepath
+    """
+    #Read the baseline conf file data/daos_server_baseline.yml
+    try:
+        with open('{}/{}'.format(basepath,
+                                 DEFAULT_YAML_FILE), 'r') as read_file:
+            default_value = yaml.safe_load(read_file)
+    except Exception as excpn:
+        print("<SERVER> Exception occurred: {0}".format(str(excpn)))
+        traceback.print_exception(excpn.__class__, excpn, sys.exc_info()[2])
+        raise ServerFailed("Failed to Read {}/{}".format(basepath,
+                                                         DEFAULT_YAML_FILE))
+
+    #Read the values from avocado_testcase.yaml file if test ran with Avocado.
+    avocado_yaml_value = ""
+    if "AVOCADO_TEST_DATADIR" in os.environ:
+        avocado_yaml_file = str(os.environ["AVOCADO_TEST_DATADIR"]).\
+                                split(".")[0] + ".yaml"
+        avocado_yaml_file_tmp = '{}.tmp'.format(avocado_yaml_file)
+
+        # Yaml Python module is not able to read !mux so need to
+        # remove !mux before reading.
+        remove_mux_from_yaml(avocado_yaml_file, avocado_yaml_file_tmp)
+
+        # Read avocado test yaml file.
+        try:
+            with open('{}'.format(avocado_yaml_file_tmp), 'r') as read_file:
+                avocado_yaml_value = yaml.safe_load(read_file)
+        except Exception as excpn:
+            print("<SERVER> Exception occurred: {0}".format(str(excpn)))
+            traceback.print_exception(excpn.__class__, excpn, sys.exc_info()[2])
+            raise ServerFailed("Failed to Read {}"
+                               .format('{}.tmp'.format(avocado_yaml_file)))
+        #Remove temporary file
+        os.remove(avocado_yaml_file_tmp)
+    #Update values from avocado_testcase.yaml in DAOS yaml variables.
+    for key in avocado_yaml_value['server_config']:
```

Comment:
A comment here about what you are doing with these two files to end up with a new result would be useful.

Additionally, the clarification of the names `default_value` and `avocado_yaml_value` as mentioned previously would help here.

Reply:
Changing the name